packages/taquito-sapling/src/sapling-tx-builder/sapling-transactions-builder.ts

```diff
@@ -0,0 +1,352 @@
+import blake from 'blakejs';
+import { secretBox } from '@stablelib/nacl';
+import { defaultMemo, KDF_KEY, OCK_KEY } from '../constants';
+import { SaplingForger } from '../sapling-forger/sapling-forger';
+import { BigNumber } from 'bignumber.js';
+import {
+  Input,
+  SaplingTransactionOutput,
+  SaplingTransactionInput,
+  ParametersBindingSig,
+  ChosenSpendableInputs,
+  ParametersOutputDescription,
+  ParametersCiphertext,
+  Ciphertext,
+  SaplingContractDetails,
+  SaplingTransaction,
+} from '../types';
+import { SaplingDiffResponse } from '@taquito/rpc';
+import { b58cdecode, Prefix, prefix } from '@taquito/utils';
+import { TzReadProvider } from '@taquito/taquito';
+import { convertValueToBigNumber } from '../sapling-tx-viewer/helpers';
+import { SaplingState } from '../sapling-state/sapling-state';
+import { SaplingWrapper } from './sapling-module-wrapper';
+import { InMemorySpendingKey } from '../sapling-keys/in-memory-spending-key';
+
+export class SaplingTransactionBuilder {
+  #inMemorySpendingKey: InMemorySpendingKey;
+  #saplingForger: SaplingForger;
+  #contractAddress: string;
+  #saplingId: string | undefined;
+  #memoSize: number;
+  #readProvider: TzReadProvider;
+  #saplingWrapper: SaplingWrapper;
+  #chainId: string | undefined;
+  #saplingState: SaplingState;
+
+  constructor(
+    inMemorySpendingKey: InMemorySpendingKey,
+    saplingForger: SaplingForger,
+    saplingContractDetails: SaplingContractDetails,
+    readProvider: TzReadProvider,
+    saplingWrapper = new SaplingWrapper()
+  ) {
+    this.#saplingForger = saplingForger;
+    this.#contractAddress = saplingContractDetails.contractAddress;
+    this.#memoSize = saplingContractDetails.memoSize;
+    this.#inMemorySpendingKey = inMemorySpendingKey;
+    this.#saplingState = new SaplingState(32);
+    this.#saplingId = saplingContractDetails.saplingId;
+    this.#saplingWrapper = saplingWrapper;
+    this.#readProvider = readProvider;
+  }
+
+  async createShieldTx(
```

Comment:
could the methods in these files use descriptions?

Reply:
I don't intend this file to be used by the users directly, so I didn't put descriptions. The separation there was mostly to help me write the unit tests.